llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1176,8 +1180,11 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
   // Enable contextual profiling instrumentation.
   const bool IsCtxProfGen = !IsPGOInstrGen && IsPreLink &&
                             PGOCtxProfLoweringPass::isContextualIRPGOEnabled();
+  const bool IsCtxProfUse = !UseCtxProfile.empty() && !PGOOpt &&
+                            Phase == ThinOrFullLTOPhase::ThinLTOPreLink;
```

Comment:
Just use the IsPreLink flag like you did above for IsCtxProfGen? What happens for regular LTO, which is the other case where IsPreLink could be true? If not supported there then consider adding an assert.

Reply:
But `IsPreLink` is actually "Is not post-link". `None` also passes. With the current way, we also exclude the full LTO case, IIUC.

We could rename `IsPreLink` because it's really `IsNotPostThinLink`, wdyt?

Reply:
Yes I think it means essentially not post link. But I'm wondering why you check IsPreLink for IsCtxProfGen above and the more restrictive check here.

Reply:
The instrumentation phase can happen without thinlto.